### PR TITLE
FUSETOOLS-1886 - do not determine camel version for runtimes other th…

### PR DESCRIPTION
**Pull request by lhein**
…an JBoss Fuse

### CODE REVIEW

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/pages/FuseIntegrationProjectWizardRuntimeAndCamelPage.java [hunk 1]
```diff
@@ -325,24 +332,40 @@ private void validate() {
...
+	
+	/**
+	 * checks whether the given runtime embeds camel libraries
+	 * 
+	 * @param runtime	the runtime to check
+	 * @return	true if the runtime contains camel libs
+	 */
+	private boolean isRuntimeContainingCamel(IRuntime runtime) {
+		// atm we only support the fuse runtime
+		return runtime.getRuntimeType().getId().startsWith("org.fusesource.ide.fuseesb.runtime.");
```
Comment (apupier) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/pages/FuseIntegrationProjectWizardRuntimeAndCamelPage.java]: What do you think of creating a IFuseToolingConstants.SERVER_FUSE_PREFIX ?

**Comment by lhein** [comment]
testPR

**Comment by lhein** [comment]
testPR

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/maven/CamelProjectConfigurator.java [hunk 2]
```diff
@@ -214,13 +215,24 @@ private void installCamelFacet(IFacetedProject fproj, IFacetedProjectWorkingCopy
...
 	}
 
 	private IProjectFacetVersion getCamelFacetVersion(String camelVersionString) throws CoreException {
 		String[] vparts = camelVersionString.split("\\.");
 		if (vparts.length>1) {
-			return camelFacet.getVersion(String.format("%s.%s", vparts[0], vparts[1]));
+			try {
+				return camelFacet.getVersion(String.format("%s.%s", vparts[0], vparts[1]));
+			} catch (IllegalArgumentException ex) {
+				// ignore
```
Comment (apupier) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/maven/CamelProjectConfigurator.java]: would be nice to have a trace in log for that

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/pages/FuseIntegrationProjectWizardRuntimeAndCamelPage.java [hunk 3]
```diff
@@ -298,19 +327,38 @@ private void validate() {
...
 				camelInfoText.setText("");
-			}
+			}			
+		}
+
+		try {
+			getWizard().getContainer().run(false, false, new IRunnableWithProgress() {
+				@Override
+				public void run(IProgressMonitor monitor) throws InvocationTargetException, InterruptedException {
+					monitor.beginTask("Resolving Apache Camel version...", IProgressMonitor.UNKNOWN);
```
Comment (apupier) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/pages/FuseIntegrationProjectWizardRuntimeAndCamelPage.java]: Externalize string

**Comment by apupier** [comment]
I have a strange behavior after selecting a Wildfly runtime
Button next and finish are disabled sometimes and enabled sometimes but Next is nto working. I have no trace.

image

image
image

**Comment by lhein** [comment]
odd...seems to be focus related behavior. When the version combo gets the focus it disables the buttons until it loses the focus. Seems to be a leftover from when I checked the camel version being available remotely.

**Comment by lhein** [comment]
testPR

**Comment by lhein** [comment]
@apupier feel free to review and merge

**Pull request merged**